ffcx/codegeneration/ufcx.h

```diff
@@ -134,6 +155,7 @@ extern "C"
     ufcx_tabulate_tensor_complex64* tabulate_tensor_complex64;
     ufcx_tabulate_tensor_complex128* tabulate_tensor_complex128;
 #endif // __STDC_NO_COMPLEX__
+    ufcx_tabulate_tensor_cuda* tabulate_tensor_cuda;
```

Comment:
Should this be included via a macro only if a cuda compiler is used?

Reply:
Right now tabulate_tensor_cuda contains a string with the tabulate tensor source code, which can be called at runtime to generate source code for NVRTC, so a cuda compiler isn't currently required for the ffcx output. . . .

Reply:
I'm not an expert on CUDA so tell me if I am wrong.

So these wrappers are for runtime compilation.

I still think it's best practice to guard code which shouldn't be seen by certain compilers using macros. Does NVRTC define anything? I found https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html for NVCC.

On that note, is anything additionally possible with NVCC?

Reply:
I've had a think about this and I see your point - the standard compiler compiles a C++ code which then calls NVRTC at runtime? In this case I simply think that needs clarification that this is all related to NVRTC.

Reply:
That's right, the wrapper is compiled as C++ code. Then user code invokes NVRTC at runtime which processes the string returned by the wrapper. I'll go ahead and clarify this.